test/libscap/CMakeLists.txt

```diff
@@ -44,9 +44,24 @@ set(LIBSCAP_TESTS_LIBRARIES "${GTEST_LIB}" "${GTEST_MAIN_LIB}" "${CMAKE_THREAD_L
 set(LIBSCAP_TESTS_DEPENDENCIES gtest scap)
 
 # Test suite asserting against pure userspace components
-file(GLOB_RECURSE USERSPACE_TEST_SUITE "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/*.cpp")
+file(GLOB_RECURSE USERSPACE_TEST_SUITE "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/*.cpp"
+	 "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/*.c"
+)
 set(LIBSCAP_TESTS_SOURCES ${USERSPACE_TEST_SUITE})
 
+# Add _GNU_SOURCE definition for makedev and set C flags
+add_definitions(-D_GNU_SOURCE)
+set_source_files_properties(
+	${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/linux/scap_fds_impl.c
+	PROPERTIES COMPILE_FLAGS "-D_GNU_SOURCE -include sys/sysmacros.h" COMPILE_DEFINITIONS
+																	  "HAVE_SYS_SYSMACROS_H"
+)
+
+# Add scap_engine_util for Linux tests
+if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+	list(APPEND LIBSCAP_TESTS_LIBRARIES scap_engine_util)
```

Comment:
Why was this line moved above?

Reply:
Fixed in https://github.com/falcosecurity/libs/pull/2271/commits/3effc75f9fef475771f6eb25b4cfa716fa49caf3.